react/BaseOverlay.tsx

```diff
@@ -105,24 +133,38 @@ export default function Popover(props: Props) {
     }
   }, [closeOnScroll, scrollBehavior, open])
 
-  const handleClick = useCallback(
-    (e: React.MouseEvent) => {
+  const handleClick: React.MouseEventHandler<HTMLDivElement> = useCallback(
+    e => {
       if (triggerMode === 'click') {
         // The Trigger works as a toggle, so if this event
-        // go up to the Trigger it will close the Popover
+        // go up to the Trigger it will close the Overlay
         e.stopPropagation()
         e.preventDefault()
       }
+
+      if (onClick) {
+        onClick(e)
+      }
     },
-    [triggerMode]
+    [onClick, triggerMode]
+  )
+
+  const popperClasses = useMemo(
+    () => ({
+      container: classes.popper,
+      backdrop: classes.backdrop,
+      popperArrow: classes.popperArrow,
+      backdropContainer: classes.backdropContainer,
+    }),
+    [classes]
```

Comment:
I don't know if this `useMemo` is doing much here. You're not creating new objects, but you're adding a method call.

Reply:
The idea of this use memo avoids the re-render in the inner components because all these classes objects are generated based on what is receiving from its parents.

Reply:
👍